x-pack/qa/security-tools-tests/src/test/java/org/elasticsearch/xpack/security/enrollment/tool/BaseRunAsSuperuserCommandTests.java

```diff
@@ -0,0 +1,297 @@
+        assertThat(terminal.getOutput(), is(emptyString()));
+    }
+
+    public void testUnhealthyCluster() throws Exception {
+        URL url = new URL(client.getDefaultURL());
+        HttpResponse healthResponse =
+            new HttpResponse(HttpURLConnection.HTTP_OK, Map.of("status", randomFrom("red")));
+        when(client.execute(anyString(), eq(clusterHealthUrl(url)), anyString(), any(SecureString.class), any(CheckedSupplier.class),
+            any(CheckedFunction.class))).thenReturn(healthResponse);
+        UserException e = expectThrows(UserException.class, this::execute);
+        assertThat(e.exitCode, equalTo(ExitCodes.UNAVAILABLE));
+        assertThat(e.getMessage(), containsString("RED"));
+        assertThat(terminal.getOutput(), is(emptyString()));
+        String error = terminal.getErrorOutput();
+        assertThat(error, stringContainsInOrder("Failed to determine the health of the cluster. Cluster health is currently RED.",
+            "This means that some cluster data is unavailable and your cluster is not fully functional.",
+            "The cluster logs (https://www.elastic.co/guide/en/elasticsearch/reference/"
+                + Version.CURRENT.major + "." + Version.CURRENT.minor + "/logging.html)" +
+                " might contain information/indications for the underlying cause"));
+        assertNoUsers();
+        assertNoUsersRoles();
+    }
+
+    public void testUnhealthyClusterWithForce() throws Exception {
+        URL url = new URL(client.getDefaultURL());
+        HttpResponse healthResponse =
+            new HttpResponse(HttpURLConnection.HTTP_OK, Map.of("status", randomFrom("red")));
+        when(client.execute(anyString(), eq(clusterHealthUrl(url)), anyString(), any(SecureString.class), any(CheckedSupplier.class),
+            any(CheckedFunction.class))).thenReturn(healthResponse);
+        execute("-f");
+        assertThat(terminal.getOutput(), is(emptyString()));
+        assertThat(terminal.getErrorOutput(), is(emptyString()));
+        assertNoUsers();
+        assertNoUsersRoles();
+    }
+
+    public void testWillRetryOnUnauthorized() throws Exception {
+        URL url = new URL(client.getDefaultURL());
+        HttpResponse unauthorizedResponse =
+            new HttpResponse(HttpURLConnection.HTTP_UNAUTHORIZED, Map.of());
+        when(client.execute(anyString(), eq(clusterHealthUrl(url)), anyString(), any(SecureString.class), any(CheckedSupplier.class),
+            any(CheckedFunction.class))).thenReturn(unauthorizedResponse);
+        UserException e = expectThrows(UserException.class, () -> execute("--verbose"));
+        String verboseOutput = terminal.getOutput();
+        assertThat(verboseOutput.split("\\n").length, equalTo(5));
+        assertThat(verboseOutput,
+            containsString("Unexpected http status [401] while attempting to determine cluster health. Will retry at most"));
```

Comment:
>Will retry at most

What does this mean? It feels rather cryptic regarding how many times we'll retry (or when).

Reply:
This is not the entire message, this is a test that matches a substring of the returned output. The message is here : https://github.com/elastic/elasticsearch/pull/74892/files#diff-5d2a8f56d638c853ae9181cb18a99a7bf20c45c1b1953e8f3a923abfa7eabbd7R231-R233